### PR TITLE
fix/flow_src: add missing method call

### PR DESCRIPTION
**Pull request by Fraser999**

### CODE REVIEW

File: src/flows_src.rs [hunk 1]
```diff
@@ -56,7 +56,10 @@ impl<'a> TopLevelSrc<'a> {
     }
 
     fn check_get_node_to_relocate(&mut self) {
-        if let (Some(candidate), false) = (self.0.action.get_node_to_relocate(), false) {
+        if self.0.action.has_relocating_node() {
```
Comment (jeanphilippeD) [inline src/flows_src.rs]: I would probably have kept the tuple check it seemed closer to the flow.
Can you add a test like other existing one for this.

Comment (Fraser999) [reply src/flows_src.rs]: I added the test.

I think the order of the parts of the `if` statement is an implementation detail, and this code is more readable to me.  I can change the order in the html doc if you prefer?

Comment (jeanphilippeD) [reply src/flows_src.rs]: fine either way.

File: src/scenario_tests.rs [hunk 2]
```diff
@@ -1271,6 +1271,31 @@ mod src_tests {
...
+            &[TestEvent::SetWorkUnitEnoughToRelocate(YOUNG_ADULT_205).to_event()],
+        );
+
+        run_test(
+            "Additional WorkUnitIncrement does not trigger a new relocate if one started",
+            &initial_state,
+            &[
+                ParsecVote::WorkUnitIncrement.to_event(),
+                TestEvent::SetWorkUnitEnoughToRelocate(NODE_ELDER_130).to_event(),
+                ParsecVote::WorkUnitIncrement.to_event(),
```
Comment (jeanphilippeD) [inline src/scenario_tests.rs]: You probably could have kept these 2 in the arrange_initial_state.

```
                ParsecVote::WorkUnitIncrement.to_event(),
                TestEvent::SetWorkUnitEnoughToRelocate(NODE_ELDER_130).to_event(),
```

What we test is only the last one

Comment (Fraser999) [reply src/scenario_tests.rs]: Yeah, I did have that originally, but I forgot we purge the events accumulated before we call `run_test()` though, so it makes the state we're asserting on not as useful.  We'd need to add more events presumably to check we have still got the right candidate, or dive into the member nodes of `action` to check which one(s) are set as relocating.

Comment (jeanphilippeD) [reply src/scenario_tests.rs]: The way I see test in genera, is, we have another test, that I think already exists, that tell us what we expect for the previous event. here we are just testing the last event.

Comment (Fraser999) [reply src/scenario_tests.rs]: I agree in general.  But in this case that's insufficient.  If I moved the 2 suggested events into the initial state, then the assert would simply be:

```rust
&AssertState {
    action_our_events: vec![
        ActionTriggered::WorkUnitIncremented.to_event(),
    ],
},
```

Maybe I'm not following what you mean, but I don't think that's enough to ensure that the second `WorkUnitIncrement` didn't overwrite the adult with the elder as the relocating node, or marking them both for relocation.

Comment (Fraser999) [reply src/scenario_tests.rs]: I understand your point now and have pushed another fixup.  Thanks :)

**Pull request merged**